src/App.js

```diff
@@ -1,16 +1,30 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatEntry from './components/ChatEntry.js';
+import ChatLog from './components/ChatLog.js';
+
+
 
 const App = () => {
+const [likesCount, setLikesCount] = useState(0);
```

Comment:
While this version of state rather cleverly avoids having a state variable of all the chat entries. There is a slight downside, though...

Right now all the entries in messages.json have `liked` set to `false`, but if we were to imagine that rather taking this from static data we would get this data from an API, it would be possible for some of the entries to have already been liked and thus have a value of `true` for `liked`.

However, with this code, those likes would not appear in this number, and in fact, if you could unlike the message, the number would go negative. (In this case you can't "unlike" that message because the `true` value just gets ignored as I'll explain in a comment in `ChatEntry`.)

Of course, there's a simple solution to that, and that's to count the likes before making your call to `useState`, but at the same time, if we _were_ taking in a list of messages from an API, we probably _would_ have to actually maintain the chat entries using `useState`.

And in that case, rather than having two state variables, one for the entries and one for the count of likes, it would likely be better to only maintain the one for the entries as the number of likes can be determined from the entries.

Having redundant state like that is a common source for bugs as the state that _should_ stay in sync falls out of sync. Not necessarily enough to be a bug, but often what a reviewer would call a "code smell".

But we're well in hypothetical world at this point, just wanted to point that out. I think for our case your implementation is fine!